### PR TITLE
I add an optional feature to copy the raw .ipynb file to the output. This allows you to include a link to download the notebook in the rendered html.

### PR DESCRIPTION
**Pull request by fredcallaway**

### CODE REVIEW

**Comment by danielfrg** [comment]
Thanks for the PR. I like the idea a lot.

Can you make a small change to make the notebook go to the same path as the output `*.html` file of the article instead of `output/notebook` I feel thats a more sane default.

File: markup.py [hunk 1]
```diff
@@ -125,6 +126,14 @@ def read(self, filepath):
 
         ignore_css = True if 'IPYNB_IGNORE_CSS' in self.settings.keys() else False
         content = fix_css(content, info, ignore_css=ignore_css)
+        if self.settings.get('IPYNB_NB_OUTPUT'):
+            target_dir = os.path.dirname(filepath).split('content/')[1]
+            title = metadata['title'].lower().replace(' ', '-')
+            nb_path = '{}/{}.ipynb'.format(target_dir, title)
+            if not os.path.isdir('output/' + target_dir):
```
Comment (danielfrg) [inline markup.py]: Replace the hardcoded `output/` to use `self.settings.get('OUTPUT_PATH')`.

**Comment by danielfrg** [comment]
Thanks for the PR!

**Comment by leemengtaiwan** [comment]
@fredcallaway Great work!
Just a potential issue: I have to change 

```
target_dir = os.path.dirname(filepath).split('content/')[1]
```

to 

```
target_dir = os.path.dirname(filepath).split('content')[1]
```

written in `markup.py` in order to make it work manually.

Maybe there is a more robust way to handle the path string handling.

**Comment by danielfrg** [comment]
Maybe using `os.basedir`?

**Comment by danielfrg** [comment]
I changed this a little bit to support any path to write on using a variable. Check the readme if you are interested.

**Comment by leemengtaiwan** [comment]
@danielfrg Thanks for the work, it become very flexible and work like a charm👍 

Just a quick thought:
Although one can check the Pelican documentation to see how `ARTICLE_SAVE_AS` work and set their own `IPYNB_NB_SAVE_AS`, it would be easier for people to understand how it work and customize their `IPYNB_NB_SAVE_AS` if there is a default value.

Something like

```
IPYNB_NB_SAVE_AS = 'notebooks/{slug}.ipynb'
```

in the documentation.

**Pull request merged**